landlab/components/species_evolution/species_evolver.py

```diff
@@ -0,0 +1,673 @@
+    with the ``dt`` parameter of the ``run_one_step`` method.
+
+    The geographic ranges of the taxa at the current model time are evaluated
+    during the ``run_one_step`` method. Taxa persistence or extinction is
+    determined by the taxon objects. Creation of child taxa is also determined
+    by these objects. Taxa metadata can be accessed with the attribute,
+    ``taxa_data_frame``.
+
+    Taxa are automatically assigned unique identifiers, ``uid``. Identifiers
+    are used to reference and retrieve taxon objects. Identifiers are assigned
+    in the order taxa are introduced to SpeciesEvolver.
+
+    The development of this component was inspired by SEAMLESS (Spatially
+    Explicit Area Model of Landscape Evolution by SimulationS). See Albert et
+    al., 2017, Systematic Biology 66.
+
+    Examples
+    --------
+    The evolution of a lowland taxa lineage in response to mountain range
+    formation is simulated using ZoneTaxon managed by ZoneController. Mountain
+    range formation is forced without processes for simplicity in this example.
+
+    Import modules used in the following examples.
+
+    >>> from landlab import RasterModelGrid
+    >>> from landlab.components import SpeciesEvolver
+    >>> from landlab.components.species_evolution import ZoneController
+
+    Create a model grid with mountain scale resolution. The elevation is
+    equally low throughout the grid at model onset.
+
+    >>> mg = RasterModelGrid((3, 7), 1000)
+    >>> z = mg.add_ones('topographic__elevation', at='node')
+    >>> z.reshape(mg.shape)
+    array([[ 1.,  1.,  1.,  1.,  1.,  1.,  1.],
+           [ 1.,  1.,  1.,  1.,  1.,  1.,  1.],
+           [ 1.,  1.,  1.,  1.,  1.,  1.,  1.]])
+
+    Instantiate the component with the grid.
+
+    >>> se = SpeciesEvolver(mg)
+
+    ZoneController requires a function that returns a mask of the total extent
+    of taxa habitat. The mask is a boolean array where `True` values represent
+    nodes that satisfy habitat conditions. Zone objects are not created here.
+    The mask only maps the extent where taxa can exist. This function returns
+    `True` where elevation is below 100, which is where the simulated lowland
+    taxa of this model can inhabit.
+
+    >>> def zone_func(grid):
+    ...     return grid.at_node['topographic__elevation'] < 100
+
+    Instantiate ZoneController with the grid and zone function. The initial
+    zones are created at controller instantiation. In this example, one zone is
+    created because all nodes of the zone mask are adjacent to each other.
+
+    >>> zc = ZoneController(mg, zone_func)
+    >>> len(zc.zones) == 1
+    True
+
+    Additional examples of controller usage are provided in ZoneController
+    documentation.
+
+    The ``mask`` of the zone is True where the conditions of the zone function
+    are met. All nodes of the grid are included because the elevation of each
+    node is below 100.
+
+    >>> zc.zones[0].mask
```

Comment:
Why is it `zones[0]`? is the `[0]` because you could index the different zones if different ones existed?

Reply:
Yes, that's all. I added some text to explain.